go/pkg/design_doc.md

```diff
@@ -0,0 +1,172 @@
+# Design Doc for PaddlePaddle TrainingJob k8s CRD
```

Comment:
Design docs must put under `docs/design`

Reply:
Thanks for your advice, I will put it under docs/design.